### ISSUE TITLE
indent rule: Invalid indentation for array in sequencing function calls

### ISSUE THREAD

**Issue by ikokostya** [eslint/eslint#6670]
**What version of ESLint are you using?**

`2.13.1`

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**
- Operation system `Linux arch 4.5.4-1-ARCH #1 SMP PREEMPT Wed May 11 22:21:28 CEST 2016 x86_64 GNU/Linux`
- Node.js version `v4.4.3`

**What did you do? Please include the actual source code causing the issue.**

Source code:

``` js
function f() {    
    return asyncCall()    
        .then(    
            'some string',    
            [                                                                                       
                1,    
                2,    
                3    
            ]    
        );    
}  
```

Run:

```
$  ./node_modules/.bin/eslint --env es6  --no-eslintrc --rule 'indent: [2, 4]' test.js 

/home/kostya/test.js
  5:13  error  Expected indentation of 8 space characters but found 12   indent
  6:17  error  Expected indentation of 12 space characters but found 16  indent
  7:17  error  Expected indentation of 12 space characters but found 16  indent
  8:17  error  Expected indentation of 12 space characters but found 16  indent
  9:13  error  Expected indentation of 8 space characters but found 12   indent

✖ 5 problems (5 errors, 0 warnings)
```

**What did you expect to happen?**

No errors.

**What actually happened? Please include the actual, raw output from ESLint.**

See above.

**Comment by platinumazure** [comment]
Thanks for the issue!

Not sure if this is working as designed or not. Marked as a bug for now-- ESLint team, feel free to change to enhancement as needed.

**Comment by oatkiller** [comment]
👍 Thanks for checking this out. I'm having the same issue.

**Comment by mattcaldwell** [comment]
👍

**Comment by peluja1012** [comment]
👍

**Comment by eomuraliev** [comment]
👍

**Comment by jamestbrown** [comment]
👍

**Comment by platinumazure** [comment]
@eslint/eslint-team Is this a bug or working as designed? I honestly don't know. There hasn't been much movement here.

**Comment by sstern6** [comment]
@platinumazure working on something i believe is related, https://github.com/eslint/eslint/issues/4161#issuecomment-247742813.

Digging into `indent.js` looks like eslint is having issues indenting with arrays on multiple lines.  I have found a few viable ways to approach this but there are a few issues open related to this I am not sure how deep to go, and depending on the approach starts breaking multiple existing tests and each case is different.  Looks like theres issues with ArrayExpressions indenting with, nested arrays, objects, and string literals from what I can see.

@ikokostya could you please provide your eslint config file to see how you are using `indent` and so I can trying working with it locally?

A few seemingly related issues:

https://github.com/eslint/eslint/issues/4161
https://github.com/eslint/eslint/issues/6007
https://github.com/eslint/eslint/issues/3737

**Comment by ikokostya** [comment]
@sstern6 

> @ikokostya could you please provide your eslint config file to see how you are using indent and so I can trying working with it locally?

https://github.com/ymaps/codestyle/blob/5b379066a8f525f1c5d5e4838699ea0a716cfcaf/packages/eslint-config-loris/es5.js#L95
